bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -714,16 +719,19 @@ private ContentPolicy getContentPolicy(@NotNull Resource imageResource) {
     private Map<String, Integer> getTransformationMap(List<String> selectorList, Resource component) throws IllegalArgumentException {
         Map<String, Integer> selectorParameterMap = new HashMap<>();
         int width = this.defaultResizeWidth;
+        boolean isAsset = DamUtil.isAsset(component);
         if (selectorList.size() > 1) {
             String widthString = (selectorList.size() > 2 ? selectorList.get(2) : selectorList.get(1));
             try {
                 width = Integer.parseInt(widthString);
                 if (width <= 0) {
                     throw new IllegalArgumentException();
                 }
-                List<Integer> allowedRenditionWidths = getAllowedRenditionWidths(component);
-                if (!allowedRenditionWidths.contains(width)) {
-                    throw new IllegalArgumentException("The requested width is not allowed in the content policy or no default");
+                if (!isAsset) {
```

Comment:
For DAM asset direct calls the challenge I see here is the `component` is that DAM asset, not the image component on the page. There is no content policy availbale.

Reply:
I agree with Vlad, this is a concern for me. Especially in features that can't be disabled. 

I don't know the requirements, so these suggestions might not have any validity.
Is there a way for the component that is rendering the image delegate to an image component? In that way the URL is still the resource URL of the component and not the direct dam asset?

Or can the path of the component that is displaying the image be passed to the adaptive image servlet so that if the request is directly for an asset, there must be a referenced component for which we can get the policy and use that policy instead?

Kyle

Reply:
@ky940819 while I agree with you and Vlad on the requirement I don't see a real solution here. The use-case is to address DAM assets directly, the adaptive image servlet will be called with a DAM asset URL (like /content/dam/....). With such an URL the servlet has no access to the content policy.

I will look into passing the path of the component that is displaying the image to the servlet. Which I think can be done, however, most likely via a query parameter. That is suboptimal for Dispatcher caching the image request.

Reply:
Alternately, we could somehow define a `default` image policy and use that for DAM-based URLs. WDYT?

Reply:
I think the adaptive image servlet is already really complex. Adding a bunch of if/else to handle different situations makes me think that this is doing two different things, although the goal is similar. 

Perhaps there should be a different servlet for this, one that is scoped to image resource types, another for DAM Assets. Even if they share some of the same back-end utility, it would be clearer.

Or even a filter for asset requests that constructs a synthetic resource with the dam asset path as the file reference, sets the resource type, and then forwards the request to the adaptive image servlet such that the logic for direct dam access can be in a single place, and the adaptive image servlet can focus solely on rendering image component resources.

Reply:
I did some experiments using the `contentPolicyDelegatePath` request parameter to provide access to the content policy. Unfortunately, this pattern only works for the image component itself, but not for components that embed the image via an image resource wrapper (like teaser or carousel). Here the component path is resolved to the original component (e.g. teaser component) and the policy of that component is used. Hence `allowedRenditionWidths` etc. which are props of the image policy are not available.